probe-rs-target/src/chip_family.rs

```diff
@@ -366,6 +368,38 @@ impl ChipFamily {
 
         Ok(())
     }
+
+    fn validate_rtt_scan_regions(&self) -> Result<(), String> {
+        for variant in &self.variants {
+            let Some(ranges) = &variant.rtt_scan_ranges else {
+                return Ok(());
+            };
+
+            let ram_regions = variant
+                .memory_map
+                .iter()
+                .filter_map(MemoryRegion::as_ram_region);
+
+            // The custom ranges must all be enclosed by exactly one of
+            // the defined RAM regions.
+            // TODO: we should allow consecutive RAM regions
```

Comment:
If the memory is two distinct ranges, just add them both? I don't see why we need to allow consecutive necessarily, but fair :)

Reply:
this todo means to allow the control block to be partially in one region, and partially in the consecutive one, because why not